### PR TITLE
refactor: remove `catalog/publish/products` meteor method, use `publishProductsToCatalog` GQL Mutation instead

### PR DESCRIPTION
**Pull request by kieckhafer**
Resolves parts of #5529   
Impact: **minor**  
Type: **refactor**

## Issue
`publishContainer` and `productGridContainer` still use the `catalog/publish/products` meteor method to publish products, which in turn uses `getGraphQLContextInMeteorMethod`.

## Solution
Re-write anywhere that calls `catalog/publish/products` to use the `publishProuctsToCatalog` mutation instead, remove the `catalog/publish/products` meteor method, which will remove `getGraphQLContextInMeteorMethod` in this instance.

## Breaking changes
Any custom code that uses the `catalog/publish/products` Meteor method will need to be re-written to use the `publishProductsToCatalog` graphql mutation

## Testing
1. See the `catalog/publish/products` meteor method has been removed
1. Attempt to publish changes to a product to the catalog.
1. See you are successful

### CODE REVIEW

File: imports/plugins/included/product-variant/containers/productGridContainer.js [hunk 1]
```diff
@@ -58,12 +58,16 @@ const wrapComponent = (Comp) => (
...
-        if (result) {
-          Alerts.toast(i18next.t("admin.catalogProductPublishSuccess", { defaultValue: "Product published to catalog" }), "success");
-        } else if (error) {
-          Alerts.toast(error.message, "error");
+    handlePublishProducts = async (productIds, mutation) => {
+      // we need to encode the productIds here to pass them to GraphQL
+      const productIdObjects = productIds.map((productId) => (
+        { namespace: "Product", id: productId }
+      ));
+      const [opaqueProductIds] = await getOpaqueIds(productIdObjects);
```
Comment (aldeed) [inline imports/plugins/included/product-variant/containers/productGridContainer.js]: Same comment here.

**Comment by kieckhafer** [comment]
@aldeed updated array destruction

@mikemurray Updated the code to get rid of that weird extra ID thing we were seeing

**Comment by kieckhafer** [comment]
I'm not sure what to do about this DCO issue. I fixed it locally, pushed, but it's still showing up. Now, when I look at my commits locally, I don't even see that commit, but all my commits are signed off:

1__reaction__less_

**Comment by kieckhafer** [comment]
Closing in favor of #5541

**Pull request closed without merging**